src/Application/GameStates/VideoState.cpp

```diff
@@ -0,0 +1,73 @@
+#include "VideoState.h"
+
+#include <Engine/Engine.h>
+#include <Engine/EngineGlobals.h>
+#include <Engine/Time/Timer.h>
+#include <Media/Audio/AudioPlayer.h>
+#include <Media/MediaPlayer.h>
+
+#include <GUI/GUIWindow.h>
+
+VideoState::VideoState(std::string_view videoFileName) : _videoFileName(videoFileName) {
+}
+
+FSMAction VideoState::enter() {
+    _skipVideo = false;
+    if (engine->config->debug.NoVideo.value()) {
+        return FSMActionTransition("videoEnd");
+    }
+
+    _movie = pMediaPlayer->loadFullScreenMovie(_videoFileName.c_str());
+    if (!_movie) {
+        return FSMActionTransition("videoEnd");
+    }
+
+    // Stop the event timer and audio before playing a video
+    pEventTimer->setPaused(true);
+    pAudioPlayer->pauseLooping();
+    pAudioPlayer->MusicPause();
+
+    // Also hide the mouse cursor
+    platform->setCursorShown(false);
+
+    // Wish we could get rid of this type of screen states
+    _previousScreenType = current_screen_type;
+    current_screen_type = SCREEN_VIDEO;
```

Comment:
Once we bring all the states of the game to the FSM system I strongly believe we can get rid of many of those cases. Or at least remove a lot of them. Right now I'm forced to keep them because we're using both systems but I'm very positive they will go away. Probably I should write a `TODO` to clarify that comment cause it's not just a wish :D